tests/test_data_utils/test_downsample_cells.py

```diff
@@ -1,98 +1,150 @@
+import sys
+import os
+sys.path.append(os.path.dirname(os.path.realpath(__file__)) + "/../../src")
 import unittest
 import pandas as pd
-import numpy as np
 from spac.data_utils import downsample_cells
 
 
 class TestDownsampleCells(unittest.TestCase):
 
     def setUp(self):
-        """Create a sample dataframe for testing."""
-        self.df = pd.DataFrame({
+        """Create sample dataframes for testing."""
+        # Random datasets for general functionality tests
+        self.df_single_random = pd.DataFrame({
             'annotation': ['annotation1'] * 50 +
                           ['annotation2'] * 30 +
                           ['annotation3'] * 20,
-            'value': np.random.rand(100)
+            'value': list(range(1, 101))
+        })
+        animals_random = [
+            'a1', 'a1', 'a1', 'a2', 'a2', 'a3', 'a4', 'a5', 'a6', 'a7'
+        ]
+        regions_random = [
+            'r1', 'r1', 'r2', 'r2', 'r2', 'r3', 'r4', 'r5', 'r6', 'r7'
+        ]
+        values_random = list(range(1, 11))
+        self.df_multi_random = pd.DataFrame({
+            'animal': animals_random,
+            'region': regions_random,
+            'value': values_random
+        })
+
+        # Fixed datasets for stratification frequency tests
+        self.df_single_fixed = pd.DataFrame({
+            'annotation': ['annotation1'] * 6 +
+                          ['annotation2'] * 3 +
+                          ['annotation3'],
+            'value': list(range(1, 11))  # Simple values from 1 to 10
+        })
+        self.df_multi_fixed = pd.DataFrame({
+            'animal': [
+                'a1', 'a1', 'a1', 'a1', 'a1', 'a2', 'a2', 'a2', 'a3', 'a4'
+            ],
+            'region': [
+                'r1', 'r1', 'r1', 'r2', 'r2', 'r1', 'r1', 'r2', 'r3', 'r4'
+            ],
+            'value': list(range(1, 11))  # Simple values from 1 to 10
         })
-        self.annotation = 'annotation'
+
+        self.annotation_single = 'annotation'
+        self.annotation_multi = ['animal', 'region']
 
     def test_annotation_existence(self):
```

Comment:
@fangliu117 , if you use the util function for checking annotation, we can remove this unit test.

Reply:
Since util function didn't apply, currently, I keep it here, but will be happy to change if needed.